### PR TITLE
BUG: fix+test DTA/TDA/PA add/sub Index

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Looking forward to doing a couple of test-cleanup passes.

### CODE REVIEW

File: pandas/tests/arithmetic/test_datetime64.py [hunk 1]
```diff
@@ -2249,6 +2249,23 @@ def test_add_datetimelike_and_dti(self, addend, tz):
 
     # -------------------------------------------------------------
 
+    def test_dta_add_sub_index(self, tz_naive_fixture):
+        # Check that DatetimeArray defers to Index classes
+        dti = date_range("20130101", periods=3, tz=tz_naive_fixture)
+        dta = dti._data
```
Comment (jreback) [inline pandas/tests/arithmetic/test_datetime64.py]: use .array

File: pandas/tests/arithmetic/test_period.py [hunk 2]
```diff
@@ -1013,6 +1013,18 @@ def test_parr_add_sub_td64_nat(self, box_transpose_fail):
         with pytest.raises(TypeError):
             other - obj
 
+    # ---------------------------------------------------------------
+    # Unsorted
+
+    def test_parr_add_sub_index(self):
+        # Check that PeriodArray defers to Index on arithmetic ops
+        pi = pd.period_range("2000-12-31", periods=3)
+        parr = pi._data
```
Comment (jreback) [inline pandas/tests/arithmetic/test_period.py]: same

File: pandas/tests/arithmetic/test_timedelta64.py [hunk 3]
```diff
@@ -480,6 +480,25 @@ def test_timedelta(self, freq):
         tm.assert_index_equal(result1, result4)
         tm.assert_index_equal(result2, result3)
 
+    def test_tda_add_sub_index(self):
+        # Check that TimedeltaArray defers to Index on arithmetic ops
+        tdi = TimedeltaIndex(["1 days", pd.NaT, "2 days"])
+        tda = tdi._data
```
Comment (jreback) [inline pandas/tests/arithmetic/test_timedelta64.py]: same

**Review by jreback — APPROVED** [review]
style comment in the test, otherwise lgtm (can do in a followup if you want), lmk.

**Comment by jbrockmendel** [comment]
_data -> array edits made

**Comment by jreback** [comment]
lgtm ping on green.

**Comment by jbrockmendel** [comment]
ping

**Comment by jreback** [comment]
thanks!

**Pull request merged**